ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/PaywallBackground.kt

```diff
@@ -1,34 +1,45 @@
 package com.revenuecat.purchases.ui.revenuecatui.composables
 
+import BlurTransformation
 import androidx.compose.foundation.layout.BoxScope
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.BlurredEdgeTreatment
-import androidx.compose.ui.draw.alpha
-import androidx.compose.ui.draw.blur
 import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
+
 import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
-import com.revenuecat.purchases.ui.revenuecatui.extensions.conditional
+import kotlin.math.roundToInt
 
 @Composable
 internal fun BoxScope.PaywallBackground(templateConfiguration: TemplateConfiguration) {
+    val transformation = if (templateConfiguration.configuration.blurredBackgroundImage)
+        BlurTransformation(context = LocalContext.current, radius = BackgroundUIConstants.blurSize.toFloatPx(),
+            scale = BackgroundUIConstants.blurScale)
+    else null
+
     templateConfiguration.images.backgroundUri?.let {
         RemoteImage(
             urlString = it.toString(),
             modifier = Modifier
-                .matchParentSize()
-                .conditional(templateConfiguration.configuration.blurredBackgroundImage) {
-                    // TODO-PAYWALLS: backwards compatibility for blurring
-                    blur(BackgroundUIConstants.blurSize, edgeTreatment = BlurredEdgeTreatment.Unbounded)
-                        .alpha(BackgroundUIConstants.blurAlpha)
-                },
```

Comment:
didn't really want to remove the modifier because it's really nice but it was easier to just apply transformations since you can get the image as a bitmap easily